lib/spack/spack/hooks/sbang.py

```diff
@@ -23,6 +25,8 @@
 else:
     shebang_limit = 127
 
+interpreter_regex = re.compile(b'#![ \t]*?([^ \t\0\n]+?)([ \t\0\n]|$)')
```

Comment:
Uh, not sure why I did `([...]|$)`. That group can just be removed. I think this was because I started out with just `[ \t\0\n]` and later realized there may just be an interpreter and then end of file. Let me fix.
```suggestion
interpreter_regex = re.compile(b'#![ \t]*?([^ \t\0\n]+)')
```